### ISSUE TITLE
Investigate best approach for communication (queues, notification - pub sub model) between the API and microservices based on AWS technologies

### ISSUE THREAD

**Issue by emettely** [bbc/digital-paper-edit-api#1]

**Comment by emettely** [comment]
```
                                                 +-------+
                                                 | Video |
                                            +----+ Pre   |
                                            |    +-------+
                                            |
+----------+            AWS Queue           |    +-------+
|          | PUSH  +-+-+-+-+-+-+-+-+-+-+    |    | Audio |
|   API    +------>+ | | | | | | | | | +<--------+ Strip +----+
|          |       +---+-+-+-+-+-+-+-+-+    |    +-------+    |
|          |         ^                      |                 |
+----------+         |                      |    +-------+    |
                     |                      |    | STT   |    |
                     |                      +----+       |    |
                     |                           +-------+    |
                     |                                        |
                     |                                        |
                     +----------------------------------------+
                                       PUSH
```

**Comment by emettely** [comment]
```
                                                                  +-------+
                                                                  | Video |
                                                             +--->+ Pre   |
                                                             |    +-------+
                                                             |
+----------+            AWS Queue              +------+      |    +-------+
|          | PUSH  +-+-+-+-+-+-+-+-+-+-+  POLL | msg  | POST |    | Audio |
|   API    +------>+ | | | | | | | | | +<------+ proc +---------->+ Strip +---+
|          |       +---+-+-+-+-+-+-+-+-+       +------+      |    +-------+   |
|          |         ^                                       |                |
+----------+         |                                       |    +-------+   |
                     |                                       |    | STT   |   |
                     |                                       +--->+       |   |
                     |                                            +-------+   |
                     |                                                        |
                     |                                                        |
                     +--------------------------------------------------------+
                                       PUSH
```

**Comment by emettely** [comment]
Alternatively, get rid of all the queues. 

```
                       +-------+
                       | Video |
                  +--->+ Pre   |
                  |    +-------+
+----------+      |
|          |      |    +-------+
|   API    | POST |    | Audio |
|          +---------->+ Strip |
|          |      |    +-------+
+----------+      |
                  |    +-------+
                  |    | STT   |
                  +--->+       |
                       +-------+
```

**Comment by pietrop** [comment]
This is a very good read on [`Microservices architecture` style](https://docs.microsoft.com/en-us/azure/architecture/guide/architecture-styles/microservices)

But to flash this out a bit more we should also look into other [`Architecture styles`](https://docs.microsoft.com/en-us/azure/architecture/guide/architecture-styles)

**TL;DR**: we don’t have to use microservices - we could use 
-  [`Web-Queue-Worker architecture style`](https://docs.microsoft.com/en-us/azure/architecture/guide/architecture-styles/web-queue-worker)
-  [`Event-Driven Architectures `](https://docs.microsoft.com/en-us/azure/architecture/guide/architecture-styles/event-driven)

**Comment by emettely** [comment]
![20190621_162914](https://user-images.githubusercontent.com/8188418/59934049-94fef500-9442-11e9-8415-ccda059bd6cb.jpg)

**Comment by emettely** [comment]
Will write up an ADR around this with an updated diagram.

**Comment by emettely** [comment]
Based on above architecture, updated
* https://github.com/bbc/digital-paper-edit-client/issues/2
* https://github.com/bbc/digital-paper-edit-stt-proxy/issues/6
* https://github.com/bbc/digital-paper-edit-audio-converter/issues/1
